# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../cool-retro-term/blinking-cursor.patch
# Copyright (C) 2020 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

From 9960b25dff271249fc774f222be799a2d87c36cb Mon Sep 17 00:00:00 2001
From: Kristian <ga53jey@mytum.de>
Date: Mon, 8 Jul 2019 15:21:17 +0200
Subject: [PATCH] Add support for blinking cursor

---
 app/qml/ApplicationSettings.qml  |  7 ++++++-
 app/qml/PreprocessedTerminal.qml |  1 +
 app/qml/SettingsTerminalTab.qml  | 18 ++++++++++++++++++
 3 files changed, 25 insertions(+), 1 deletion(-)

diff --git a/app/qml/ApplicationSettings.qml b/app/qml/ApplicationSettings.qml
index 11d4330..412af96 100644
--- a/app/qml/ApplicationSettings.qml
+++ b/app/qml/ApplicationSettings.qml
@@ -59,6 +59,8 @@ QtObject{
     property real burnInQuality: 0.5
     property bool useFastBurnIn: Qt.platform.os === "osx" ? false : true
 
+    property bool blinkingCursor: true
+
     onWindowScalingChanged: handleFontChanged();
 
     // PROFILE SETTINGS ///////////////////////////////////////////////////////
@@ -208,7 +210,8 @@ QtObject{
             burnInQuality: burnInQuality,
             useCustomCommand: useCustomCommand,
             customCommand: customCommand,
-            useFastBurnIn: useFastBurnIn
+            useFastBurnIn: useFastBurnIn,
+            blinkingCursor: blinkingCursor
         }
         return stringify(settings);
     }
@@ -296,6 +299,8 @@ QtObject{
         customCommand = settings.customCommand !== undefined ? settings.customCommand : customCommand
 
         useFastBurnIn = settings.useFastBurnIn !== undefined ? settings.useFastBurnIn : useFastBurnIn;
+
+        blinkingCursor = settings.blinkingCursor !== undefined ? settings.blinkingCursor : blinkingCursor
     }
 
     function loadProfileString(profileString){
diff --git a/app/qml/PreprocessedTerminal.qml b/app/qml/PreprocessedTerminal.qml
index d0d9cee..cb8698c 100644
--- a/app/qml/PreprocessedTerminal.qml
+++ b/app/qml/PreprocessedTerminal.qml
@@ -94,6 +94,7 @@ Item{
         smooth: !appSettings.lowResolutionFont
         enableBold: false
         fullCursorHeight: true
+        blinkingCursor: appSettings.blinkingCursor
 
         session: QMLTermSession {
             id: ksession
diff --git a/app/qml/SettingsTerminalTab.qml b/app/qml/SettingsTerminalTab.qml
index 8cefdb8..914a90c 100644
--- a/app/qml/SettingsTerminalTab.qml
+++ b/app/qml/SettingsTerminalTab.qml
@@ -114,6 +114,24 @@ Tab{
                 }
             }
         }
+        GroupBox{
+            title: qsTr("Cursor")
+            Layout.fillWidth: true
+            ColumnLayout {
+                anchors.fill: parent
+                CheckBox{
+                    id: blinkingCursor
+                    text: qsTr("Blinking Cursor")
+                    checked: appSettings.blinkingCursor
+                    onCheckedChanged: appSettings.blinkingCursor = checked
+                }
+                Binding{
+                    target: blinkingCursor
+                    property: "checked"
+                    value: appSettings.blinkingCursor
+                }
+            }
+        }
         GroupBox{
             title: qsTr("Colors")
             Layout.fillWidth: true
From 8c1293e985f70688c7190e30a87ad1af26ad6a90 Mon Sep 17 00:00:00 2001
From: Kristian Klemon <kristian.klemon@gmail.com>
Date: Wed, 6 May 2020 16:39:48 +0200
Subject: [PATCH] Add blinkingCursor property (#25)

Co-authored-by: Kristian <ga53jey@mytum.de>
---
 lib/TerminalDisplay.cpp | 3 +++
 lib/TerminalDisplay.h   | 2 ++
 2 files changed, 5 insertions(+)

diff --git a/lib/TerminalDisplay.cpp b/lib/TerminalDisplay.cpp
index 04d54c0..87a384e 100644
--- a/qmltermwidget/lib/TerminalDisplay.cpp
+++ b/qmltermwidget/lib/TerminalDisplay.cpp
@@ -1302,6 +1302,9 @@ void TerminalDisplay::showResizeNotification()
 
 void TerminalDisplay::setBlinkingCursor(bool blink)
 {
+  if (_hasBlinkingCursor != blink)
+      emit blinkingCursorStateChanged();
+      
   _hasBlinkingCursor=blink;
 
   if (blink && !_blinkCursorTimer->isActive())
diff --git a/lib/TerminalDisplay.h b/lib/TerminalDisplay.h
index 3535756..d8371d8 100644
--- a/qmltermwidget/lib/TerminalDisplay.h
+++ b/qmltermwidget/lib/TerminalDisplay.h
@@ -99,6 +99,7 @@ class KONSOLEPRIVATE_EXPORT TerminalDisplay : public QQuickPaintedItem
 
    Q_PROPERTY(bool enableBold           READ getBoldIntense   WRITE setBoldIntense NOTIFY boldIntenseChanged )
    Q_PROPERTY(bool fullCursorHeight     READ fullCursorHeight WRITE setFullCursorHeight NOTIFY fullCursorHeightChanged)
+   Q_PROPERTY(bool blinkingCursor       READ blinkingCursor   WRITE setBlinkingCursor NOTIFY blinkingCursorStateChanged)
    Q_PROPERTY(bool antialiasText        READ antialias       WRITE setAntialias)
    Q_PROPERTY(QStringList availableColorSchemes READ availableColorSchemes NOTIFY availableColorSchemesChanged)
 
@@ -615,6 +616,7 @@ public slots:
     void availableColorSchemesChanged();
     void colorSchemeChanged();
     void fullCursorHeightChanged();
+    void blinkingCursorStateChanged();
     void boldIntenseChanged();
 
 protected:
